lib/config.js

```diff
@@ -812,8 +821,9 @@ util.parseFile = function(fullFilename) {
     throw new Error("Cannot parse config file: '" + fullFilename + "': " + e3);
   }
 
-  // Keep track of this configuration sources, including empty ones
-  if (typeof configObject === 'object') {
+  // Keep track of this configuration sources, including empty ones, unless the skipConfigSources flag is set to true in the options
+  var skipConfigSources = options !== undefined && options.skipConfigSources === true;
```

Comment:
This pattern is repeated several times, and given the nature of the optional options object it's likely it will be repeated in the future.  Do you think this could be condensed into a method that handles these edge cases and can be maintained in one place?

Reply:
@lorenwest Sure. What kind of signature do you have in mind? :)

Reply:
Sure, something like this:
```
const skipConfigSources = getOption('skipConfigSources', true);
```
The method could be defined within `loadFileConfigs()` for now

Reply:
All right, I'll get started on it, shouldn't take long. Since it's used in loadFileConfigs *and* in parseFile, I'll put it a bit higher, though.
getOption(optionName: string, defaultValue) ?

Reply:
Sounds good, and since it's outside of the function you'll have to pass the options object as arg1.
```
getOption(optionObj: object, optionName: string, defaultValue: any);
```
make sense?

Reply:
obviously :D.

Reply:
@lorenwest done :)